app/handlers.py

```diff
@@ -56,13 +57,39 @@ def join_iac(data, expected_length=12):
         return combined
 
     @staticmethod
-    def validate_case(case_json):
-        if not case_json.get("active", False):
-            raise InactiveCaseError
+    def validate_iac_active(iac_json, case_json):
+        if not iac_json.get("active", False):
+            try:
+                if case_json['caseGroup']['caseGroupStatus'] in ['COMPLETE', 'COMPLETEDBYPHONE']:
+                    logger.info('Attempt to use inactive iac for completed case')
+                    raise CompletedCaseError
+            except KeyError:
+                logger.warn("Field case_json['caseGroup']['caseGroupStatus'] not found")
+
+            logger.info('Attempt to use inactive iac for incomplete case')
+            raise InactiveIACError
+
+    def check_case_sample_unit_type_valid(self, case_json):
+        try:
+            assert case_json['sampleUnitType'] == 'H'
```

Comment:
Best not to use asserts, under certain options the interpreter can ignore the statement completely.

Reply:
@AdamHawtin this already existed prior to these changes and is a valid use case for this (should only be skipped if certain optimisation flags were used at runtime) 👍